DocumentsFromSnapshotMigration/src/test/java/org/opensearch/migrations/bulkload/CustomTransformationTest.java

```diff
@@ -0,0 +1,282 @@
+
+            var snapshotCreator = new CreateSnapshot(args, testSnapshotContext.createSnapshotCreateContext());
+            snapshotCreator.run();
+
+            esSourceContainer.copySnapshotData(tempDirSnapshot.toString());
+
+            int exitCode;
+            int finalExitCodeCount = 0;
+            int runs = 0;
+            do {
+                exitCode = processRunner.apply(new RunData(tempDirSnapshot, tempDirLucene, osTargetContainer));
+                runs++;
+                if (exitCode == eventualExitCode) {
+                    finalExitCodeCount++;
+                }
+                log.atInfo().setMessage("Process exited with code: {}").addArgument(exitCode).log();
+                // Clean tree for subsequent run
+                deleteTree(tempDirLucene);
+            } while (finalExitCodeCount < eventualExitCodeCount && runs < initialExitCodes * 2);
+
+            // Assert doc count on the source and target cluster match expected
+            validateFinalClusterDocs(
+                esSourceContainer,
+                osTargetContainer,
+                DocumentMigrationTestContext.factory().noOtelTracking(),
+                expectedSourceDocs,
+                expectedTargetDocs
+            );
+        } finally {
+            deleteTree(tempDirSnapshot);
+        }
+    }
+
+    private static String createIndexNameTransformation(String existingIndexName, String newIndexName) {
+        JSONArray rootArray = new JSONArray();
+        JSONObject firstObject = new JSONObject();
+        JSONArray jsonConditionalTransformerProvider = new JSONArray();
+
+        // JsonJMESPathPredicateProvider object
+        JSONObject jsonJMESPathPredicateProvider = new JSONObject();
+        jsonJMESPathPredicateProvider.put("script", String.format("index._index == '%s'", existingIndexName));
+        JSONObject jsonJMESPathPredicateWrapper = new JSONObject();
+        jsonJMESPathPredicateWrapper.put("JsonJMESPathPredicateProvider", jsonJMESPathPredicateProvider);
+        jsonConditionalTransformerProvider.put(jsonJMESPathPredicateWrapper);
+
+        JSONArray transformerList = new JSONArray();
+
+        // First JsonJoltTransformerProvider
+        JSONObject firstJoltTransformer = new JSONObject();
+        JSONObject firstJoltScript = new JSONObject();
+        firstJoltScript.put("operation", "modify-overwrite-beta");
+        firstJoltScript.put("spec", new JSONObject().put("index", new JSONObject().put("\\_index", newIndexName)));
+        firstJoltTransformer.put("JsonJoltTransformerProvider", new JSONObject().put("script", firstJoltScript));
+        transformerList.put(firstJoltTransformer);
+
+        jsonConditionalTransformerProvider.put(transformerList);
+        firstObject.put("JsonConditionalTransformerProvider", jsonConditionalTransformerProvider);
+        rootArray.put(firstObject);
+        return rootArray.toString();
+    }
+
+    @SneakyThrows
+    private static int runProcessAgainstTarget(
```

Comment:
It looks like this was basically copied from the ProcessLifecycleTest.  Can you please find a way to share/re-use rather than copy?

Reply:
I've done a good amount of refactoring so this should be different, with some shared code move into parent class